### ISSUE TITLE
Convert `PrimUndefined` case from `BackendExpr`

### ISSUE THREAD

**Issue by JordanMartinez** [purescm/purescm#123]
Using `Data.Ring` as an example...
```purs
instance ringRecordCons ::
  ( IsSymbol key
  , Row.Cons key focus subrowTail subrow
  , RingRecord rowlistTail row subrowTail
  , Ring focus
  ) =>
  RingRecord (RL.Cons key focus rowlistTail) row subrow where
  subRecord _ ra rb = insert (get ra - get rb) tail
    where
    insert = unsafeSet key :: focus -> Record subrowTail -> Record subrow
    key = reflectSymbol (Proxy :: Proxy key)
    get = unsafeGet key :: Record row -> focus
    tail = subRecord (Proxy :: Proxy rowlistTail) ra rb
```

you'll see `prim-undefined` printed below. I believe it's supposed to be a placeholder argument. So, perhaps converting it the smallest value we can provide would work? Maybe via `genSym`? Maybe just `#t`?
```scheme
  (scm:define ringRecordCons
    (scm:lambda (dictIsSymbol0)
      (scm:lambda (_)
        (scm:lambda (dictRingRecord2)
          (scm:let ([semiringRecordCons13 (((Data.Semiring.semiringRecordCons dictIsSymbol0) prim-undefined) ((scm:hashtable-ref dictRingRecord2 "SemiringRecord0" #f) prim-undefined))])
            (scm:lambda (dictRing4)
              (scm:let ([semiringRecordCons25 (semiringRecordCons13 ((scm:hashtable-ref dictRing4 "Semiring0" #f) prim-undefined))])
                (scm:letrec* (($record (scm:make-hashtable scm:string-hash scm:string=?))) (scm:hashtable-set! $record "subRecord" (scm:lambda (v6)
                  (scm:lambda (ra7)
                    (scm:lambda (rb8)
                      (scm:let*
                        ([key9 ((scm:hashtable-ref dictIsSymbol0 "reflectSymbol" #f) Type.Proxy.Proxy)]
                         [get10 (Record.Unsafe.unsafeGet key9)])
                          (((Record.Unsafe.unsafeSet key9) (((scm:hashtable-ref dictRing4 "sub" #f) (get10 ra7)) (get10 rb8))) ((((scm:hashtable-ref dictRingRecord2 "subRecord" #f) Type.Proxy.Proxy) ra7) rb8))))))) (scm:hashtable-set! $record "SemiringRecord0" (scm:lambda (_)
                  semiringRecordCons25)) $record))))))))
```

**Comment by f-f** [comment]
I believe this case should never happen, because if it does then there's something wrong in the typeclass dictionary? The haskell implementation would just throw an error - see [the Transpiler module](https://github.com/purescm/purescm/blob/e2c3e336c61e62dca9013bfc8d95ad9d0f1e98f2/legacy/src/Language/PureScript/Scheme/CodeGen/Transpiler.hs#L229) and the corresponding [test](https://github.com/purescm/purescm/blob/master/legacy/test/resources/scheme/PureScheme.Test.Let.LambdasInFunction/lib.sls)

**Comment by purefunctor** [comment]
> I believe this case should never happen, because if it does then there's something wrong in the typeclass dictionary?

I'm not sure why we'd want to throw an error here? The `undefined` seems to be to eliminate an abstraction over the instance dictionary, for example, given this test case:
```purescript
-- @inline Snapshot.LambdasInFunction.foo never
module Snapshot.LambdasInFunction where

import Prelude

foo :: forall a. Ring a => a -> a -> a
foo x y =
  let
    add _ _ = x + y
    sub _ _ = x - y
  in
    add (sub x y) (add x y)

foo' :: Int
foo' = foo 1 1
```

The following Scheme code is produced, assuming that `prim-undefined` generates `#f`
```scheme
(library
  (Snapshot.LambdasInFunction lib)
  (export
    foo
    foo$p)
  (import
    (prefix (chezscheme) scm:)
    (prefix (_Chez_Runtime lib) rt:)
    (prefix (Data.Ring lib) Data.Ring.))

  (scm:define foo
    (scm:lambda (dictRing0)
      (scm:hashtable-ref ((scm:hashtable-ref dictRing0 "Semiring0" #f) #f) "add" #f)))

  (scm:define foo$p
    (((foo Data.Ring.ringInt) 1) 1)))
```

And this is how `ringInt` is defined:
```scheme
  (scm:define ringInt
    (scm:letrec* (($record (scm:make-hashtable scm:string-hash scm:string=?))) (scm:hashtable-set! $record "sub" intSub) (scm:hashtable-set! $record "Semiring0" (scm:lambda (_)
      Data.Semiring.semiringInt)) $record))
```

**Comment by JordanMartinez** [comment]
Fixed by #128
